flake-modules/tests.nix

```diff
@@ -52,6 +72,6 @@
         package-options = pkgs.callPackage ../tests/package-options.nix { inherit nixvimConfiguration; };
 
         lsp-all-servers = pkgs.callPackage ../tests/lsp-servers.nix { inherit nixvimConfiguration; };
-      } // import ../tests { inherit pkgs pkgsUnfree helpers; };
+      } // callTests ../tests { };
```

Comment:
Should this commit go further and incorporate the full changes from [`01ef05e` (#2104)](https://github.com/nix-community/nixvim/pull/2104/commits/01ef05e140731acd9a81b2a70a67259aa140b411)?

Maybe I'll pull this out into another PR, but if I do it'll block this PR.

Reply:
I've opened #2433 to tackle this refactoring separately.